src/generators.nix

```
@@ -1,5 +1,45 @@
 { lib, deploy }:
 {
+  /**
+    Synopsis: mkDeployNodes _nixosConfigurations_
+
+    Generate the `nodes` attribute expected by deploy-rs
+    where _nixosConfigurations_ are `nodes`.
+    **/
+  mkDeployNodes = nixosConfigurations:
+    let
+      op = _: config:
+        let
+          activate = deploy.lib.${config.config.nixpkgs.system}.activate;
+          hm-profiles = (n: v: {
+            user = n;
+            profilePath = "/nix/var/nix/profiles/per-user/${n}/home-manager";
+            path = activate.nixos.home-manager v.home;
+          });
+        in
+        {
+          hostname = config.config.networking.hostName;
+          profiles =
+            {
+              system = {
+                user = "root";
+                path = activate.nixos config;
+              };
+            }
+            //
+            (
+              builtins.mapAttrs
+                hm-profiles
+                config.home-manager.users
```

```suggestion
                config.system.build.homes or config.home-manager.users
```
We should use the custom build here too, since that adds in the registry and allows for hm-only configuration

---

This is better resolved via https://github.com/divnix/digga/pull/84